pkg/kubelet/prober/testing/fake_manager.go

```diff
@@ -43,7 +44,7 @@ func (FakeManager) CleanupPods(_ map[types.UID]sets.Empty) {}
 func (FakeManager) Start() {}
 
 // UpdatePodStatus simulates updating the Pod Status.
-func (FakeManager) UpdatePodStatus(_ types.UID, podStatus *v1.PodStatus) {
+func (FakeManager) UpdatePodStatus(_ types.UID, podStatus *kubecontainer.PodStatus, _ sets.Set[string]) {
 	for i := range podStatus.ContainerStatuses {
 		podStatus.ContainerStatuses[i].Ready = true
```

Comment:
This is not sufficient to test the new update strategy. I need to add the started too here.

Reply:
actually maybe not... let's see after refactoring `generateAPIPodStatus`